web-api/src/embedding.rs

```diff
@@ -49,7 +49,11 @@ pub(crate) struct Embedder {
 
 impl Embedder {
     pub(crate) fn run(&self, s: &str) -> Result<Embedding, InternalError> {
-        self.bert.run(s).map_err(InternalError::from_std)
+        let embedding = self.bert.run(s);
+        match embedding {
```

Comment:
:art: we could match the result right away without an intermediate assignment
```suggestion
        match self.bert.run(s) {
```